drake/solvers/Optimization.h

```diff
@@ -130,7 +130,7 @@ namespace Drake {
     virtual ~QuadraticConstraint() {}
 
     virtual void eval(const Eigen::Ref<const Eigen::VectorXd>& x, Eigen::VectorXd& y) const override { y.resize(getNumConstraints()); y=.5*x.transpose()*Q*x + b.transpose()*x; }
-    virtual void eval(const Eigen::Ref<const TaylorVecXd>& x, TaylorVecXd& y) const override { y.resize(getNumConstraints()); y = .5*x.transpose()*constantTaylorVecXd(Q)*x + constantTaylorVecXd(b).transpose()*x; };
+    virtual void eval(const Eigen::Ref<const TaylorVecXd>& x, TaylorVecXd& y) const override { y.resize(getNumConstraints()); y = .5*x.transpose()* Q.cast<TaylorVarXd>() * x + b.cast<TaylorVarXd>().transpose() * x; };
```

Comment:
that's fine.
i actually think we should implement (in EigenUtils.h, or wherever) our own support for multiplying double matrices with autodiff matrices (from #1312).  it's ridiculuously inefficient to have to allocate a new autodiff matrix, looping through an zeroing all of the gradients, every time we want to do this.